src/textual/app.py

```diff
@@ -985,7 +985,7 @@ async def run_auto_pilot(
             )
         finally:
             try:
-                if auto_pilot_task is not None:
+                if auto_pilot_task is not None and not self._exit_renderables:
```

Comment:
I don't find this a very elegant way of checking if we are shutting down already, but it works...